Sources/Host/Darwin/Platform.cpp

```diff
@@ -118,6 +118,26 @@ std::string Platform::GetThreadName(ProcessId pid, ThreadId tid) {
 const char *Platform::GetSelfExecutablePath() {
   return LibProc::GetExecutablePath(getpid());
 }
+
+ErrorCode Platform::TranslateKernError(kern_return_t kret) {
+  switch (kret) {
+  case KERN_SUCCESS:
+    return ds2::kSuccess;
+  case KERN_FAILURE:
+    return ds2::kErrorUnknown;
+  case KERN_MEMORY_FAILURE:
+  case KERN_INVALID_ADDRESS:
+    return ds2::kErrorInvalidAddress;
+  case KERN_PROTECTION_FAILURE:
```

Comment:
Interesting, I was thinking kErrorInvalidAddress was really link a BAD_PARAM where the address is just wrong.